src/main/kotlin/uk/gov/justice/digital/hmpps/manageoffencesapi/service/OffenceService.kt

```diff
@@ -27,13 +27,20 @@ class OffenceService(
 ) {
   fun findOffencesByCode(code: String): List<Offence> {
     log.info("Fetching offences by offenceCode")
-    val offences = offenceRepository.findByCodeStartsWithIgnoreCase(code).map { it ->
-      val children = offenceRepository.findByParentOffenceId(it.id)
-      transform(it, children.map { child -> child.id })
-    }
-    return offences.map {
-      it.copy(schedules = transform(offenceSchedulePartRepository.findByOffenceId(it.id)))
+    val offences = offenceRepository.findByCodeStartsWithIgnoreCase(code)
```

Comment:
refactored this method to improve performance. Replace calls to the DB in a loop with one call outside the loop